runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -0,0 +1,111 @@
+package org.corfudb.runtime.view.workflows;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.ManagementClient;
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.corfudb.runtime.exceptions.WorkflowResultUnknownException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.Sleep;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * An abstract class that defines a generec workflow request structure.
+ * <p>
+ * Created by Maithem on 1/19/18.
+ */
+@Slf4j
+public abstract class WorkflowRequest {
+
+    protected int retry;
+
+    protected Duration timeout;
+
+    protected Duration pollPeriod;
+
+    protected CorfuRuntime runtime;
+
+    /**
+     * Send a workflow request
+     *
+     * @param layout current layout
+     * @return a uuid that corresponds to the created workflow
+     */
+    protected abstract UUID sendRequest(Layout layout);
+
+    /**
+     * Select an orchestrator and return a client
+     *
+     * @param layout the layout that is used to select an orchestrator
+     * @return a management client that is connected to the selected
+     * orchestrator
+     */
+    protected abstract ManagementClient getOrchestrator(Layout layout);
```

Comment:
Implementation should probably be here.... But I'm not sure this is the right place to put it

Reply:
I moved it to the abstract class, what do you suggest?